src/main/java/com/minecolonies/coremod/colony/Colony.java

```diff
@@ -353,44 +426,158 @@ public int getDimension()
     }
 
     /**
-     * Increment the mobs killed by this colony.
-     * <p>
-     * Will award achievements for mobs killed.
+     * Sets the name of the colony.
+     * Marks dirty.
+     *
+     * @param n new name.
+     */
+    public void setName(final String n)
+    {
+        name = n;
+        markDirty();
+    }
+
+    /**
+     * Returns the world the colony is in.
+     *
+     * @return World the colony is in.
      */
-    public void incrementMobsKilled()
+    @Nullable
+    public World getWorld()
+    {
+        return world;
+    }
+
+    @Override
+    public long getDistanceSquared(@NotNull final BlockPos pos)
+    {
+        return BlockPosUtil.getDistanceSquared2D(center, pos);
+    }
+
+    @Override
+    public boolean hasTownHall()
     {
-        killedMobs++;
-        final int mobKills = this.getKilledMobs();
-        if (mobKills >= NUM_MOBS_ACHIEVEMENT_FIRST)
+        return townHall != null;
+    }
+
+    /**
+     * Returns the ID of the colony.
+     *
+     * @return Colony ID.
+     */
+    @Override
+    public int getID()
+    {
+        return id;
+    }
+
+    /**
+     * Get the amount of statistic.
+     *
+     * @param statistic the statistic.
+     * @return amount of statistic.
+     */
+    private int getStatisticAmount(@NotNull String statistic)
+    {
+        switch (statistic)
         {
-            this.triggerAchievement(ModAchievements.achievementKillOneMob);
+            case TAG_GUARD_MOBS:
+                return killedMobs;
+            case TAG_MINER_ORES:
+                return minedOres;
+            case TAG_MINER_DIAMONDS:
+                return minedDiamonds;
+            case TAG_BUILDER_HUTS:
+                return builtHuts;
+            case TAG_FISHERMAN_FISH:
+                return caughtFish;
+            case TAG_FARMER_WHEAT:
+                return harvestedWheat;
+            case TAG_FARMER_POTATOES:
+                return harvestedPotatoes;
+            case TAG_FARMER_CARROTS:
+                return harvestedCarrots;
+            case TAG_LUMBERJACK_SAPLINGS:
+                return plantedSaplings;
+            case TAG_LUMBERJACK_TREES:
+                return felledTrees;
+            default:
+                return 0;
         }
-        if (mobKills >= NUM_MOBS_ACHIEVEMENT_SECOND)
+    }
+
+    /**
+     * increment statistic amount.
+     * @param statistic the statistic.
+     */
+    private void incrementStatisticAmount(@NotNull String statistic)
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') The Cyclomatic Complexity of this method "incrementStatisticAmount" is 12 which is greater than 10 authorized. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)